src/main/java/org/visab/newgui/sessionoverview/viewmodel/NewSessionOverviewViewModel.java

```diff
@@ -291,44 +119,40 @@ public int compare(SessionStatus o1, SessionStatus o2) {
         }
     };
 
-    public SimpleStringProperty getTotalSessionsProperty() {
-        return totalSessionsProperty;
+    /**
+     * Extended getter that always adds the current sessions before returning them.
+     * 
+     * @return the (sorted) session statuses.
+     */
+    public ObservableList<SessionStatus> getSessionStatuses() {
+        sessionStatuses.clear();
+        for (SessionStatus status : WebAPI.getInstance().getSessionAdministration().getSessionStatuses()) {
+            sessionStatuses.add(status);
+        }
+        return sortSessionStatuses(sessionStatuses);
     }
```

Comment:
Okay now I understand why this is not just an observable list field.
Is there any way in which we can observe changes to the sessionStatus list in the SessionListenerAdministration?
I suppose just subscribing `SessionOpened` is not enough, because of inconsistency when clearing inactive sessions by button press, correct?

Reply:
If my last assumption is correct, I think its fine that we just call this from updateSessionGrid.
To make the way in which we intend this to work clearer thought, I would suggest removing the sessionStatus field completely and just creating a new instance every time this is called.
Furthermore, I think giving this method a name other than get would be good, since there is quite a bit of logic happening.
Maybe `querySessionStatuses` or sth.